### ISSUE TITLE
A more interactive lockscreen

### ISSUE THREAD

**Issue by JoeRandini** [peterhajas/MobileNotifier#106]
For example, I think that you should be able to touch the alert icon on the lockscreen, and it should take you directly to your messages app etc. I also think that there should be an alert on your lockscreen before you even view the text message.

**Comment by peterhajas** [comment]
Joe, I agree with the second part of your issue, but not with the first.

Contrary to popular belief, the lockscreen is not supposed to be an interactive element. If I did make it interactive, it would be very limited and non destructive (i.e. hitting it in your pocket wouldn't do anything).

However, an updated lockscreen is definitely something I'm interested in providing!

**Comment by io41** [comment]
I agree that lockscreen interaction should be non destructive, but I think JoeRandini has a point too. I would suggest that tapping an alert on the lockscreen would have the same effect as anywhere else, but would be validated by unlocking. That way it's not so frustrating to unlock and having to retrigger mobilenotifier just to tap/dismis the alert, after already having viewed mobilenotifier on the lockscreen.

In summary, what I'm suggesting in the lockscreen is:

Tap mobilenotifier. --> alerts are shown.
Tap desired alert. --> alert is highlighted, with "unlock to view" message, and cancel "X".
Unlock --> Unlocks phone, and triggers whatever action associated with normally tapping the alert.
  or
"X" --> alert highlight is removed and screen stays locked.

**Comment by peterhajas** [comment]
I really like this idea @io41, and I'm going to check out if it's doable.

**Comment by Kaizersoze** [comment]
I think the io41 idea can be simplified.
- The notification list should always be displayed. button "MobileNotifier"has no purpose since we want, in 99% of cases, see our lockscreen missed event (the notification list view should be sorted by app priority).
- Why display an alert when you tap on a notification involving one more tap, so that to avoid destructive interactions (eg in pocket), just activate the code lock device?

In summary, what I'm suggesting in the lockscreen is:

Notification list view is always shown
Tap desired Notification --> Unlocks phone, asking for lock code if used and must be checked, and triggers whatever action associated with normally tapping the alert.

**Comment by timnovinger** [comment]
@Drudoo and I played around with this and it's not possible from what we see. Sorry.
